packages/http-client-csharp/emitter/src/lib/operation-converter.ts

```diff
@@ -166,14 +165,18 @@ function fromSdkOperationParameters(
 
 function fromSdkHttpOperationParameter(
   sdkContext: CSharpEmitterContext,
-  p: SdkPathParameter | SdkQueryParameter | SdkHeaderParameter | SdkBodyParameter,
+  p: SdkModelPropertyType,
   rootApiVersions: string[],
 ): InputParameter {
   const isContentType =
     p.kind === "header" && p.serializedName.toLocaleLowerCase() === "content-type";
   const parameterType = fromSdkType(sdkContext, p.type);
   const format = p.kind === "header" || p.kind === "query" ? p.collectionFormat : undefined;
-  const serializedName = p.kind !== "body" ? p.serializedName : p.name;
+
+  // use serializedName if available, but fallback to name
+  // special case for body as the name is incorrectly set to "body" https://github.com/Azure/typespec-azure/issues/2292
+  const serializedName =
+    "serializedName" in p && p.kind !== "body" ? (p.serializedName ?? p.name) : p.name;
```

Comment:
I think we should always use `p.name`.

Parameters and properties are both modeled as `SdkModelPropertyType` here in TCGC because typespec treats them as the same thing.
But in TCGC, `serializedName` has been deprecated on properties. Parameters have the property of `serializedName` but it actually does not really make sense unless it is a query or header parameter.
As SDK generators, we should focus on its name which represents the client name, serializedName of a parameter should be working as `wire info` like we have in properties.

Reply:
So why do we have the separate NameInRequest property?

Reply:
Sorry I misread the variable here - I thought it was `name`.
It is fine, never mind.